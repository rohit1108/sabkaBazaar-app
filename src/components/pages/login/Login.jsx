import React from "react";
import LoginBody from "../../organisms/signinBody/LoginBody"
import Navbar from "../../organisms/Navbar/Navbar";
import "./login.scss";
import Helmet from "react-helmet";

const Login =()=>{
    return(
        <>
         <Helmet>
            <meta charSet="utf-8" />
            <meta name="description" content="You can view different offers on the prducts and different categories of product." />
            <title>Login</title>
          </Helmet>
        <Navbar isAuth={false}/>
        <LoginBody/>
        </>
    )
    }
export default Login;