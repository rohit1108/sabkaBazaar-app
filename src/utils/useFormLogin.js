import { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";
import { login } from "../service/Auth";

function useForm() {
const navigate = useNavigate();
  const initialvalues = {
   
    email: "",
    password: "",
  
  };
  const [formValues, setFormValues] = useState({initialvalues});
  const [formErrors, setFormErrors] = useState({});
  const [isSubmit, setIsSubmit] = useState(false);


  const handleSubmit = (e) => {
    e.preventDefault();
    setFormErrors(validate(formValues));
    setIsSubmit(true);
    login(formValues);
  };
  const getEmail = () => {
    return localStorage.getItem("EMAIL");
};
const removeEmail = () => {
    localStorage.removeItem("EMAIL");
};

  useEffect(() => {
    if (Object.keys(formErrors).length === 0 && isSubmit) {
        navigate("/home");
        
        console.log(formValues);
    }
  }, [formErrors, formValues, navigate, isSubmit]);
  const validate = (values) => {
    const errors = {};
    const emailRegex = /[a-z0-9]+@[a-z]+.[a-z]{2,3}/;
    const passwordRegex = /^(?=.*[0-9])(?=.*[a-zA-Z])([a-zA-Z0-9]+)$/;
    
   
    if (!values.email) {
      errors.email = "Email is required";
    } else if (!emailRegex.test(values.email)) {
      errors.email = "This is not a valid email format";
    }
    if (!values.password) {
      errors.password = "Password is required";
    } else if (!passwordRegex.test(values.password)) {
      errors.password = "Credentails are incorrect";
    }
    return errors;
  };

  return {  validate, formValues,  setFormValues, formErrors, isSubmit, handleSubmit, getEmail,removeEmail };
}
export default useForm;