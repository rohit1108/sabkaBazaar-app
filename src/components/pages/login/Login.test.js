import Login from "./Login"
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import useFormLogin from "../../../utils/useFormLogin"

describe("Login credentials form on submit", () => {
    const login = {
        email: "group@sapient.com",
        password: "Password123!",
    };

    let emailInput, passwordInput, submitButton;

    beforeEach(() => {
        // eslint-disable-next-line testing-library/no-render-in-setup
        render(
                <BrowserRouter>
                    <Login/>
                </BrowserRouter>
        );

        emailInput = screen.getByLabelText(/Email/i);
        passwordInput = screen.getByLabelText(/Password/i);
        submitButton = screen.getByRole("button", { name: /Login/i });

        userEvent.clear(emailInput);
        userEvent.clear(passwordInput);
    });
    const fillAndSubmit = (useFormLogin) => {
        userEvent.type(emailInput, useFormLogin.email);
        userEvent.type(passwordInput, useFormLogin.password);
        userEvent.click(submitButton);
    };
    test("should display error when email is not filled", async () => {
        fillAndSubmit({
            ...login,
            email: "",
            
        });
        const errorMessageEl = await screen.findByText(/Email is required/i);
        expect(errorMessageEl).toMatchSnapshot();
        expect(errorMessageEl).toBeInTheDocument();
    });

    test("should display error when password is not filled", async () => {
        fillAndSubmit({
            ...login,
            password: "",
        });
        const errorMessageEl = await screen.findByText(/Password is required/i);
        expect(errorMessageEl).toMatchSnapshot();
        expect(errorMessageEl).toBeInTheDocument();
    });
});
