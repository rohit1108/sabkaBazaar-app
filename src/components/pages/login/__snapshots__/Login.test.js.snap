// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login credentials form on submit should display error when email is not filled 1`] = `
<p
  class="form-error"
>
  Email is required
</p>
`;

exports[`Login credentials form on submit should display error when password is not filled 1`] = `
<p
  class="form-error"
>
  Password is required
</p>
`;
